shell/platform/common/cpp/client_wrapper/include/flutter/texture_registrar.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_TEXTURE_REGISTRAR_H_
+#define FLUTTER_SHELL_PLATFORM_COMMON_CPP_CLIENT_WRAPPER_INCLUDE_FLUTTER_TEXTURE_REGISTRAR_H_
+
+#include <flutter_texture_registrar.h>
+
+#include <stdint.h>
+#include <memory>
+
+namespace flutter {
+
+// An external texture interface declaration.
+class Texture {
+ public:
+  virtual ~Texture() {}
+  // This is in response to the texture copy request interface, providing the
+  // |height| and |width| parameters of bounds.
+  // In some cases, we need to scale the texture to the bounds size to reduce
+  // memory usage.
+  virtual const PixelBuffer* CopyPixelBuffer(size_t width, size_t height) = 0;
```

Comment:
What is the ownership model of the return value here? If it's a copy presumably something needs to release it, but in that case I would expect a `unique_ptr`. The meaning/use of the return value needs to be clearly documented.

Reply:
This still needs to be addressed.